corehq/apps/reports/generic.py

```diff
@@ -669,8 +662,12 @@ def export_response(self):
         Returns the tabular export of the data, if available.
         """
         self.is_rendered_as_export = True
+        report_state = self.__getstate__()
+        datespan = report_state['request'].pop('datespan')
+        report_state['request_params']['startdate'] = datespan.startdate.isoformat()
+        report_state['request_params']['enddate'] = datespan.enddate.isoformat()
```

Comment:
I think I saw this in one other spot as well, in addition to "reforming the datespan" object after calling `__getstate__`. I think the dictionary returned by `__getstate__` should be usable/json serializable immediately, meaning the `__getstate__` method can handle converting the datespan into a start and end date, and the `__setstate__` method can handle creating a datespan object from a start and end date.